### PR TITLE
Github Actions

### PR DESCRIPTION
**Pull request by wershlak**
@cizra Can you take a look here? I think ideally we'd have the docker file for the mud here with the mud and CI to publish it all here.
- I removed the travis stuff and replaced it with github actions. Nice to see all the output here.
- I created a new docker hub account "sneezymuddeployer" and have it uploading. We need to invite this to the sneezy mud org and I can fix up this PR
- https://hub.docker.com/repository/registry-1.docker.io/wershlak/sneezymud2/tags?page=1&ordering=last_updated
- See my comments: this is working but I feel like the code checkout should just be in the CI and not in the docker file but I didn't get that working yet.

### CODE REVIEW

**Review by Aion2501 — APPROVED** [review]

File: .github/workflows/docker.yml
Comment (wershlak) [inline .github/workflows/docker.yml:21]: Need to fix

File: Dockerfile
Comment (wershlak) [inline Dockerfile:12]: So I'd really love to remove this but it didn't work locally to just copy the code into the container as on line 9

File: .github/workflows/test.yml
Comment (cizra) [inline .github/workflows/test.yml:21]: It'll hang, as Sneezy won't exit. Besides, you don't have the DB set up.
Would it be possible instead to use a compose file, combining DB with a MUD client logging in as admin and shutting down the MUD? Maybe also run a couple of commands. We could use this to build up a suite of blackbox tests.

File: Dockerfile
Comment (cizra) [reply Dockerfile:12]: Huh, why? It's good practice to build inside the container, as it'll stabilize your build environment (Ubuntu updates in the pipeline's worker agents won't cause build to break because Boost updated a version, or such.

**Review by cizra — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml
Comment (cizra) [inline .github/workflows/test.yml:20]: Why would you build it outside of container, if it's possible to build inside container?

**Review by cizra — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml
Comment (cizra) [reply .github/workflows/test.yml:21]: oops, it'll just compile, not run it. Disregard the bit about hanging.

File: .github/workflows/test.yml
Comment (wershlak) [reply .github/workflows/test.yml:20]: Ooh so this was me copying over the travis stuff. Apparently, that's what the travis CI was doing when we created PRs. This is a separate workflow from the container build and is just meant as a test. We could just build containers on every commit in every branch (as a way of testing that the code change is good). Think that's better?

File: .github/workflows/docker.yml
Comment (wershlak) [reply .github/workflows/docker.yml]: fixed

File: Dockerfile
Comment (wershlak) [reply Dockerfile:12]: Yeah exactly. Ok so since there might be some additional bits to work out I'm going to leave it this was and get back to it another time.

File: .github/workflows/docker.yml
Comment (wershlak) [reply .github/workflows/docker.yml]: This is for the CI container. I updated this line to specifically use ubuntu-20.04

File: .github/workflows/test.yml
Comment (wershlak) [reply .github/workflows/test.yml:20]: I updated the other things and removed the WIP. I think this is ready to merge unless you think we should do something different with this test workflow.

Comment (cizra) [reply .github/workflows/test.yml:20]: Yeah, we could build the container but avoid publishing it, so as not to spew junk into Docker Hub.

**Comment by cizra** [comment]
Why's the Travis build still required by GitHub, if you removed the .travis.yml? It's blocking the merge.

File: .github/workflows/test.yml
Comment (cizra) [inline .github/workflows/test.yml:14]: Where can I see the output of this?

**Review by cizra — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml
Comment (wershlak) [reply .github/workflows/test.yml:14]: The action says it only runs on files added in a specific commit. https://github.com/marketplace/actions/c-style-check-compiled

The output is on the "checks" tab of the PR but there's nothing interesting for this one.

I haven't tested it at all though. I'm expecting a style check could be painful for our codebase but we can try it out.

Comment (cizra) [reply .github/workflows/test.yml:14]: OK, fair enough

File: .github/workflows/test.yml
Comment (wershlak) [reply .github/workflows/test.yml:20]: I _think_ the travis stuff is still here cause it was here when I started the branch and the PR. I need to test that out.

Other than that I need to fix the test here but I'm super behind after all that time off. I'm going to merge this so we can get things publishing and fix the test when I get a chance. Feel like there's so many PRs in the pipeline 😱

**Pull request merged**
